x/evm/precompile/precompile.go

```diff
@@ -222,3 +231,25 @@ var isMutation map[PrecompileMethod]bool = map[PrecompileMethod]bool{
 
 	OracleMethod_queryExchangeRate: false,
 }
+
+// ReturnToParentGasMeter resets the ctx.GasMeter back to a parent GasMeter before precompile execution.
+// Additionally, handles the out of gas panic by resetting the gas meter and returning an error.
+// This is used in order to avoid panics and to allow for the EVM to continue cleanup if the tx or query run out of gas.
+func ReturnToParentGasMeter(ctx sdk.Context, contract *vm.Contract, parentGasMeter sdk.GasMeter, err *error) func() {
```

Comment:
Pretty sure we can simplify this by not returning an anonymous function and removing the `()` at the defer site.